src/system/ExtensionsModule/Controller/ModuleController.php

```diff
@@ -0,0 +1,193 @@
+                ];
+                $actions[] = [
+                    'url' => $this->get('router')->generate('zikulaextensionsmodule_admin_remove', ['id' => $module['id']]),
+                    'icon' => 'trash-o',
+                    'color' => '#c00',
+                    'title' => $this->__f('Uninstall %s', ['%s' => $module['displayname']])
+                ];
+                break;
+            case \ModUtil::STATE_MISSING:
+                // Nothing to do.
+                break;
+            case \ModUtil::STATE_UPGRADED:
+                $actions[] = [
+                    'url' => $this->get('router')->generate('zikulaextensionsmodule_admin_upgrade', ['id' => $module['id']]),
+                    'icon' => 'refresh',
+                    'color' => '#00c',
+                    'title' => $this->__f('Upgrade %s', ['%s' => $module['displayname']])
+                ];
+                break;
+            case \ModUtil::STATE_INVALID:
+                // nothing to do.
+                // do not allow deletion of invalid modules if previously installed (#1278)
+                break;
+            case \ModUtil::STATE_NOTALLOWED:
+                $actions[] = [
+                    'url' => $this->get('router')->generate('zikulaextensionsmodule_admin_remove', ['id' => $module['id']]),
+                    'icon' => 'trash-o',
+                    'color' => '#c00',
+                    'title' => $this->__f('Remove %s', ['%s' => $module['displayname']])
+                ];
+                break;
+            case \ModUtil::STATE_UNINITIALISED:
+            default:
+                if ($module['state'] < 10) {
+                    $actions[] = [
+                        'url' => $this->get('router')->generate('zikulaextensionsmodule_admin_initialise', ['id' => $module['id']]),
+                        'icon' => 'cog text-success',
+                        'color' => '#0c0',
+                        'title' => $this->__f('Install %s', ['%s' => $module['displayname']])
+                    ];
+                } else {
+                    $actions[] = [
+                        'url' => $this->get('router')->generate('zikulaextensionsmodule_admin_compinfo', ['id' => $module['id']]),
+                        'icon' => 'info-circle',
+                        'color' => 'black',
+                        'title' => $this->__f('Incompatible version: %s', ['%s' => $module['displayname']])
+                    ];
+                }
+                break;
+        }
+
+        if ($module['state'] != \ModUtil::STATE_INVALID) {
+            $actions[] = [
+                'url' => $this->get('router')->generate('zikulaextensionsmodule_admin_modify', ['id' => $module['id']]),
+                'icon' => 'wrench',
+                'color' => 'black',
+                'title' => $this->__f('Edit %s', ['%s' => $module['displayname']])
+            ];
+        }
+
+        return $actions;
+    }
+
+    private function isCoreModule($moduleName)
```

Comment:
Ah maybe this breaks when the module is not installed / inactive (thinking of PageLock here).